app/controllers/api/predictions_controller.rb

```diff
@@ -52,8 +70,28 @@ def build_predictions
       end
     end
 
-    def invalid_message
+    def find_prediction
+      @prediction = Prediction.find(params[:id])
+    end
+
+    def invalid_api_message
       { error: 'invalid API token', status: :unauthorized }
     end
+
+    def unauthorized_user_message
+      {
+        error: 'user is unauthorized to view this private prediction',
+        status: :unauthorized
+      }
+    end
+
+    def successfully_updated?
+      return false unless @user.authorized_for(@prediction)
```

Comment:
Also a nice to have would be that the auth check raises an exception which the top level controller catches then renders the relevant response so that all the unauthorised responses are handled in one place. 
```ruby
# Prediction API controller
def show 
   raise UnauthorisedRequest unless @user.authorized_for(@prediction)
   #... rest of action
end

#... Application controller
rescue_from UnauthorisedRequest, with: :handle_unauthorised_request

protected

def handle_unauthorised_request
  render json: {error: 'user is unauthorized to view this private prediction'}, status: :unauthorized
end 


```